File: java/singletonPattern/VoucherCheck.java
package singletonPattern;

public class VoucherCheck {
	public VoucherCheck(String id) {
		if (id == "AZ568DH")
			signIn();
		else
			System.out.println("Id:"+id+" was not recognized");
	}

	private void signIn() {
		VoucherProvider voucher = VoucherProvider.getInstance();
		System.out.println("Voucher sent");
	}
}

File: php/singletonPattern/VoucherCheck.php
<?php 
include 'VoucherProvider.php';

class VoucherCheck{

	public function __construct() {}
	
	/**
	 * Supply voucher 
	 * 
	 * @return void
	 */
	public function signIn() {
		
		$newVoucher = VoucherProvider::getVoucher();
		
	}

}


